src/main/java/io/cdap/plugin/gcp/firestore/sink/FirestoreRecordWriter.java

```diff
@@ -58,16 +59,36 @@ public class FirestoreRecordWriter extends RecordWriter<NullWritable, Map<String
    */
   public FirestoreRecordWriter(TaskAttemptContext taskAttemptContext) {
     Configuration config = taskAttemptContext.getConfiguration();
+
     String projectId = config.get(FirestoreConfig.NAME_PROJECT);
+    String databaseId = config.get(FirestoreConfig.NAME_DATABASE);
+    Boolean isServiceAccountFilePath = false;
+    String serviceAccountJson = config.get(FirestoreConfig.NAME_SERVICE_ACCOUNT_JSON);
     String serviceAccountFilePath = config.get(FirestoreConfig.NAME_SERVICE_ACCOUNT_FILE_PATH);
     String collection = Strings.nullToEmpty(config.get(FirestoreConstants.PROPERTY_COLLECTION)).trim();
+    // Get Service Account whether JSON or FilePath
+    String serviceAccountType = config.get(FirestoreConfig.NAME_SERVICE_ACCOUNT_TYPE);
+
+    LOG.debug("Initialize RecordReader(projectId={}, databaseId={}, collection={}, serviceAccountType={}, isServiceAccountFilePath={}, serviceFilePath={}, " +
+    "serviceAccountJson={}", projectId, databaseId, collection, serviceAccountType, isServiceAccountFilePath, serviceAccountFilePath, serviceAccountJson);
+    String serviceAccount = "";
+    if(serviceAccountType.equalsIgnoreCase("FilePath")) {
+      serviceAccount = config.get(FirestoreConfig.NAME_SERVICE_ACCOUNT_FILE_PATH);
+      isServiceAccountFilePath = true;
+    } else if(serviceAccountType.equalsIgnoreCase("json")) {
+      serviceAccount = config.get(FirestoreConfig.NAME_SERVICE_ACCOUNT_JSON);
+      isServiceAccountFilePath = false;
+    } else {
+      throw new FirestoreInitializationException("Service account type can only be either a File Path or JSON.");
+    }
+
     this.batchSize = config.getInt(FirestoreSinkConstants.PROPERTY_BATCH_SIZE, 25);
     this.useAutogeneratedId = config.getBoolean(FirestoreSinkConstants.PROPERTY_ID_TYPE, false);
-    LOG.debug("Initialize RecordWriter(projectId={}, collection={}, serviceFilePath={}, batchSize={}, " +
-        "useAutogeneratedId={})", projectId, collection, serviceAccountFilePath, batchSize,
-      useAutogeneratedId);
 
-    this.db = FirestoreUtil.getFirestore(serviceAccountFilePath, projectId);
+    LOG.debug("Initialize RecordWriter(projectId={}, databaseId={}, collection={}, serviceAccountType={}, isServiceAccountFilePath={}, serviceFilePath={}, " +
+    "serviceAccountJson={}, batchSize={}, useAutogeneratedId={}", projectId, databaseId, collection, serviceAccountType, isServiceAccountFilePath, serviceAccountFilePath, serviceAccountJson, batchSize, useAutogeneratedId);
```

Comment:
We should not log `serviceAccountJson` as it contains customer security configuration.